src/rawposix/src/sys_calls.rs

```diff
@@ -0,0 +1,1161 @@
+//! System syscalls implementation
+//!
+//! This module contains all system calls that are being emulated/faked in Lind.
+use crate::fs_calls::kernel_close;
+use cage::memory::vmmap::{VmmapOps, *};
+use cage::{cagetable_init, add_cage, cagetable_clear, get_cage, remove_cage, Cage, Zombie};
+use cage::signal::signal::{lind_send_signal, convert_signal_mask};
+use cage::timer::{IntervalTimer};
+use fdtables;
+use libc::sched_yield;
+use parking_lot::{RwLock, Mutex};
+use std::ffi::CString;
+use std::path::PathBuf;
+use std::sync::atomic::Ordering::*;
+use std::sync::atomic::{AtomicI32, AtomicU64};
+use std::sync::Arc;
+use std::time::Duration;
+use sysdefs::constants::err_const::{get_errno, handle_errno, syscall_error, Errno, VERBOSE};
+use sysdefs::constants::fs_const::{STDERR_FILENO, STDOUT_FILENO, STDIN_FILENO};
+use sysdefs::constants::lind_platform_const::FDKIND_KERNEL;
+use sysdefs::constants::sys_const::{EXIT_SUCCESS, DEFAULT_UID, DEFAULT_GID, SIGKILL, SIGSTOP, SIG_BLOCK, SIG_UNBLOCK, SIG_SETMASK, ITIMER_REAL};
+use sysdefs::data::fs_struct::{SigactionStruct, ITimerVal};
+use typemap::datatype_conversion::*;
+use dashmap::DashMap;
+
+
+/// Reference to Linux: https://man7.org/linux/man-pages/man2/fork.2.html
+///
+/// `fork_syscall` creates a new process (cage object). The newly created child process is an exact copy of the
+/// parent process (the process that calls fork) apart from it's cage_id and the parent_id
+/// In this function we separately handle copying fd tables and clone vmmap talbe and create a new Cage object
```

Comment:
typo on table

Reply:
I think since this function is a major departure from just calling down to the kernel probably need more in the description like where the rest of fork is handled and why were doing this, could also use some grammar cleanup